pkg/utils/common/common.go

```diff
@@ -140,6 +146,39 @@ func GenOpenAPI(inst *cue.Instance) ([]byte, error) {
 	return out.Bytes(), nil
 }
 
+// RefineParameterInstance refines cue instance to merely include `parameter` identifier
+func RefineParameterInstance(inst *cue.Instance) (*cue.Instance, error) {
+	paramVal := inst.LookupDef(velacue.ParameterTag)
+	if !paramVal.Exists() {
+		return nil, fmt.Errorf("cue identifier 'parameter' doesn't exists")
```

Comment:
we should allow component or trait without parameters. In this case, we can generate an empty data.

Reply:
It's weird that even though there is no parameter in cue.Instance, `paramVal.Exists()` still returns true. And the `Kind` of `paramVal` is `BottomKind`. 

![image](https://user-images.githubusercontent.com/5144108/121778540-df4a8500-cbc9-11eb-8a82-3b10a71242f6.png)

I can just return an parameter with empty object here so that the outer function `utils.ConvertOpenAPISchema2SwaggerObject` can get the index from the `Schema` map: 

![image](https://user-images.githubusercontent.com/5144108/121779387-f25f5400-cbcd-11eb-8ff4-8caba70661a8.png)


But now `utils.prepareParameterCue` returns an error when there is no parameter, do you expect that we still generate ConfigMap for those components/traits without parameter?

Reply:
yes, we should still generate ConfigMap even no parameter, and those workload and traits are possible to have no parameters.

Reply:
Ok, I have changed the code to make it possible to generate ConfigMap even no parameter.